File: examples/java/src/main/java/com/xebia/functional/xef/java/auto/MealPlan.java
package com.xebia.functional.xef.java.auto;

import java.util.List;
import java.util.concurrent.ExecutionException;

public class MealPlan {
    public String name;
    public List<Recipe> recipes;

    private static class Recipe {
        public String name;
        public List<String> ingredients;

        @Override
        public String toString() {
            return "Recipe{" +
                  "name='" + name + '\'' +
                  ", ingredients=" + ingredients +
                  '}';
        }
    }

    @Override
    public String toString() {
        return "MealPlan{" +
              "name='" + name + '\'' +
              ", recipes=" + recipes +
              '}';
    }

    public static void main(String[] args) throws ExecutionException, InterruptedException {
        try (AIScope scope = new AIScope()) {
            scope.prompt("Meal plan for the week for a person with gall bladder stones that includes 5 recipes.", MealPlan.class)
                  .thenAccept(mealPlan -> System.out.println(mealPlan))
                  .get();
        }
    }
}

File: example/src/main/kotlin/com/xebia/functiona/langchain4k/auto/MealPlan.kt
package com.xebia.functiona.langchain4k.auto

import com.xebia.functional.auto.ai
import kotlinx.serialization.Serializable

@Serializable
data class MealPlan(val name: String, val recipes: List<Recipe>) {
    fun prettyPrint(): String {
        return recipes.joinToString("\n") { "${it.name}:\n${it.ingredients.joinToString("\n")}" }
    }
}

suspend fun main() {
    val mealPlan: MealPlan = ai("Meal plan for the week for a person with gall bladder stones that includes 5 recipes.")
    println(
        """The meal plan for the week is: 
        |${mealPlan.prettyPrint()}""".trimMargin()
    )
}
